apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/adapters/User.ts

```diff
@@ -36,58 +55,29 @@ export class RocketChatUserAdapter {
 		if (!user) {
 			return;
 		}
-		const internalBridgedUserId = MatrixBridgedUser.getById(user._id);
-		if (!internalBridgedUserId) {
+		const internalBridgedUser = await MatrixBridgedUser.getBridgedUserByLocalId(user._id);
+		if (!internalBridgedUser) {
 			return;
 		}
-		const { mui: externalUserId, remote } = internalBridgedUserId;
+		const { mui: externalUserId, remote } = internalBridgedUser;
 
-		return this.createFederatedUserInstance(externalUserId, user, remote);
+		return createFederatedUserInstance(externalUserId, user, remote);
 	}
 
-	public async getInternalUserById(userId: string): Promise<IUser | null> {
-		return Users.findOneById(userId);
+	public async getInternalUserById(userId: string): Promise<IUser> {
+		const user = await Users.findOneById(userId);
+		if (!user || !user.username) {
+			throw new Error(`User with internalId ${userId} not found`);
+		}
+		return user;
 	}
 
 	public async createFederatedUser(federatedUser: FederatedUser): Promise<void> {
-		const existingLocalUser = await Users.findOneByUsername(federatedUser.internalReference.username || '');
+		const existingLocalUser = await Users.findOneByUsername(federatedUser.getUsername() || '');
```

Comment:
I guess we querying an empty string is not really intended right? if so, this is better if written like this:
```suggestion
		const existingLocalUser = federatedUser.getUsername() && await Users.findOneByUsername(federatedUser.getUsername());
```